### ISSUE TITLE
Documentation by Object

### ISSUE THREAD

**Issue by tnelson-doghouse** [uselagoon/lagoon-cli#348]
**Is your feature request related to a problem? Please describe.**
I often what to know what objects I can operate on, and what I can do to them

**Describe the solution you'd like**
If the docs had a page for each object (eg, "project", "user"), and listed:
- The commands for directly operating on them (eg. lagoon add project)
- The commands that link them to things (eg. lagoon add project-group)
...that'd be great.  

Each command should just have a short description (ie. one line), and link to the main command page

**Describe alternatives you've considered**
Digging through the docs like I do now.

**Comment by shreddedbacon** [comment]
One problem with that, the docs are currently automatically generated from the CLI structure using `cobra`. 

To do what you suggest would mean manual documentation changes, which we kind of want to avoid due to issues it brings.

Generally though the `short` description of commands should provide enough information to show references, for example looking at `lagoon add` the short description says which ones can be used on project and which ones are for users. We could improve these descriptions in some cases though potentially.
```
$ lagoon add
Add a project, or add notifications and variables to projects or environments

Usage:
  lagoon add [command]

Aliases:
  add, a

Available Commands:
  deploytarget               Add a DeployTarget to lagoon
  deploytarget-config        Add deploytarget config to a project
  group                      Add a group to Lagoon, or add a group to an organization
  notification               Add notifications or add notifications to projects
  organization               Add a new organization to Lagoon
  organization-administrator Add an administrator to an Organization
  organization-deploytarget  Add a deploy target to an Organization
  project                    Add a new project to Lagoon, or add a project to an organization
  project-group              Add a project to a group in lagoon
  user                       Add a user to lagoon
  user-group                 Add a user to a group in lagoon
  user-sshkey                Add an SSH key to a user
  variable            Add or update a variable to an environment or project
```

I know that isn't really ideal or solves your particular issue of having a list of commands that can be used on specific resources in Lagoon, but the alternative is potentially out of date documentation pages because the auto-generated docs won't update pages manually created.

**Comment by tnelson-doghouse** [comment]
I agree with the goal of auto-generation, but wouldn't it be possible to set up an additional automatic generation where:
- Each command is marked with an object (and the default object is "other")
- A second set of documentation is generated that pulls out those summaries and crosslinks to the original documentation
- Just possibly, a writeup on (or link to a writeup on) each object.  These writeups/links would have to be manually maintained, but hopefully this would only be relevant when adding a new object

Cobra might not do that easily (XSLT would do it, but people seem to have turned their backs on XML just because it's better at documents than at being an on-the-wire protocol).  

HTH,

**Comment by tnelson-doghouse** [comment]
Sure, fair enough.  This sounds like a Cobra problem, not an Amazee problem.  Thanks.

**Comment by shreddedbacon** [comment]
> This sounds like a Cobra problem

Well, not really. Cobra just builds the documentation based on the command structure, it does what it does and it does it well.

The problem you're describing is not a problem that I think cobra can solve. Cobra doesn't understand that one command could be linked to a specific object or not, because it doesn't exist in cobra.

**Comment by tnelson-doghouse** [comment]
So the problem is that cobra only understands commands, not objects.  Still sounds like a cobra problem :p .
